### PR TITLE
Like and Pat features

### PR DESCRIPTION
**Pull request by asawo**
**What**
- Like: keeps count of how many users have "liked" their dog. Each user can only like a dog once.
- Pat: same as Like, but users can give dogs a Pat an unlimited number of times.
- Implement mocks/stubs 

**Why**
- To have an element of interaction between users and doggos 
- Mock/stubs for testing

### CODE REVIEW

File: server/test/test-server.js [hunk 1]
```diff
@@ -34,31 +37,31 @@ describe(`Page load test for "/" endpoint`, function() {
...
 			.request(server)
 			.post('/signin')
 			.send({
 				username: 'test',
-				password: 'test'
+				password: 'test',
 			})
-			.end(function(err, res) {
+			.end(function (err, res) {
 				// expect(err).to.be.a('null')
```
Comment (VeysonD) [inline server/test/test-server.js:65]: I'm not sure if this test should still be commented out

File: server/test/test-server.js [hunk 2]
```diff
@@ -67,219 +70,222 @@ describe(`Test for "/signin" endpoint`, function() {
...
 				expect(err).to.be.a('null')
 			})
 	})
 
-	it('createAccount should create a user account', done => {
-		const formData = {
-			username: 'test1',
-			password: 'test1'
+	it('createAccount should create a user account', (done) => {
+		const objectStub = {
```
Comment (VeysonD) [inline server/test/test-server.js:91]: probably give this a better name like `userStub`

I know it's just for a simple test, but we should keep it a good habit to always strive for good variable naming

File: server/test/test-server.js [hunk 3]
```diff
@@ -67,219 +70,222 @@ describe(`Test for "/signin" endpoint`, function() {
...
+			result: null,
+			formData: { username: 'test1', password: 'test1' },
 		}
 
+		const bcryptStub = sinon.stub(dbcontroller.bcrypt, 'hash').resolves('hi')
```
Comment (VeysonD) [inline server/test/test-server.js:96]: 😆

File: server/test/test-server.js [hunk 4]
```diff
@@ -67,219 +70,222 @@ describe(`Test for "/signin" endpoint`, function() {
...
 				expect(err).to.be.a('null')
 			})
 	})
 
-	it('getDoggo should return doggo data', done => {
+	it('getDoggo should return doggo data', (done) => {
+		const doggoStub = {
+			doggoname: 'stubbyBoi',
+			description: 'stubby',
+			imgUrl: 'http://www.stubby.boi',
```
Comment (VeysonD) [inline server/test/test-server.js:227]: lol

File: server/test/test-server.js (hunk 2 above)
Comment (asawo) [reply server/test/test-server.js]: Done

File: server/test/test-server.js (hunk 1 above)
Comment (asawo) [reply server/test/test-server.js]: it shouldn't be, but I'm not sure how to solve this just yet

**Comment by VeysonD** [comment]
This PR is getting too large

It might be best to wrap things up and include it in another PR

File: server/controllers/dbcontroller.js [hunk 5]
```diff
@@ -55,7 +55,7 @@ const authenticateUser = async (user, password) => {
 
 const loadAll = async () => {
 	const result = await db.any(
-		'SELECT doggoid, doggoname, description, imageurl, username FROM doggos ORDER BY dateupdated DESC'
+		'SELECT doggos.doggoid, doggoname, description, imageurl, username, SUM(likes.likecount) AS likestotal FROM doggos INNER JOIN likes ON doggos.doggoid = likes.doggoid GROUP BY doggos.doggoid, likes.likecount'
```
Comment (VeysonD) [inline server/controllers/dbcontroller.js:58]: 👍

File: server/controllers/dbcontroller.js [hunk 6]
```diff
@@ -55,7 +55,7 @@ const authenticateUser = async (user, password) => {
 
 const loadAll = async () => {
 	const result = await db.any(
-		'SELECT doggos.doggoid, doggoname, description, imageurl, username, SUM(likes.likecount) AS likestotal FROM doggos INNER JOIN likes ON doggos.doggoid = likes.doggoid GROUP BY doggos.doggoid, likes.likecount'
+		'SELECT doggos.doggoid, doggoname, imageurl, description, username, SUM(likes.likecount) AS likestotal FROM doggos LEFT JOIN likes ON doggos.doggoid = likes.doggoid GROUP BY doggos.doggoid, likes.likecount'
```
Comment (VeysonD) [inline server/controllers/dbcontroller.js:58]: So why are you doing a left join instead of inner?

Comment (asawo) [reply server/controllers/dbcontroller.js:58]: So inner join will not show dogs with 0 likes (NULL), while left join will

**Pull request merged**